test/qunit/tests/list-of-pairs-collection-creator.js

```diff
@@ -16,6 +16,31 @@ define([
 
     module( "Galaxy client app tests" );
 
+    test( "Collection creation", function() {
+        var pcc = new PCC({
+                datasets    : DATA._1,
+                historyId   : 'fakeHistoryId'
+            }),
+            server = sinon.fakeServer.create();
+
+        var requestJSON;
+        server.respondWith( 'POST', '/api/histories/fakeHistoryId/contents/dataset_collections', function( request ){
+            requestJSON = JSON.parse( request.requestBody );
+            request.respond(
+                200,
+                { "Content-Type": "application/json" },
+                JSON.stringify({
+                    fakeResponse: 'yes'
+                })
+            );
+        });
+
+        //console.debug( 'requestBody:', JSON.stringify( requestJSON, null, '  ' ) );
+        pcc.createList( 'Heres a collection' );
+        server.respond();
+        deepEqual( requestJSON, DATA._1requestJSON );
+    });
+
```

Comment:
So:
- qunit may have reversed the order it's tests are run - it now runs like a queue and it used to run like a stack. That's why the last test evaluated first.
- It started at 2 (_.uniqueId generally starts at 1), because _.uniqueId is also used by backbone for it's cid's.
- The ids for DATA._1 were only generated once (before on test 6, now on test 2 - 1 doesn't use them) because _ensureIds would decorate the DATA._1 objects directly. All tests after the first wouldn't need ids - they were already there. 

Because the order changed: 
- test 1 ran first and _.uniqueId was used for the cid of the creator in test 1 (id -> 1)
- test 2 then ran and _.uniqueId was used for the cid of the creator in test 2 (id -> 2)
- ids were generated for ensureIds starting at the next _.uniqueId: 3
--> the actual ids were now starting at 3 changed and didn't match up with the expected.

The main issue is that this is just crappy test design on my part. Order shouldn't matter, so It shouldn't expect those ids if it's going to use the global counter in `uniqueId` - I had also forgotten that backbone uses that counter, too. And crappy code in the _ensureIds shouldn't be decorating the objects directly, but cloning them first.

In any event, like I said above - **I'm fine with the moving that to the first test**, I just wanted to find out why. Feel free to leave as is, or alternately change `datasets1` in `paired-collection-creator.data.js` to:

    var datasets1 = [
        { name: 'SET1-01_1.fastq', id: '14' },
        { name: 'SET1-01_2.fastq', id: '15' },
        { name: 'SET1-02_1.fastq', id: '12' },
        { name: 'SET1-02_2.fastq', id: '13' },
        { name: 'SET1-03_1.fastq', id: '10' },
        { name: 'SET1-03_2.fastq', id: '11' },
        { name: 'SET1-04_1.fastq', id: '8' },
        { name: 'SET1-04_2.fastq', id: '9' },
        { name: 'SET1-05_1.fastq', id: '6' },
        { name: 'SET1-05_2.fastq', id: '7' },
        { name: 'SET1-06_1.fastq', id: '4' },
        { name: 'SET1-06_2.fastq', id: '5' },
        { name: 'SET1-07_1.fastq', id: '2' },
        { name: 'SET1-07_2.fastq', id: '3' }
    ];

This will prevent the ids being generated at all.